index.js

```diff
@@ -8,6 +8,7 @@ const log = require('fancy-log');
 
 module.exports = function gulpPug(options) {
   const opts = Object.assign({}, options);
+  const namefunc = opts.name;
```

Comment:
You can remove this temporary variable. At this point you don't know if `opts.name` is a string or a function so the name of this variable is confusing.